snappy_pipeline/workflows/somatic_variant_signatures/__init__.py

```diff
@@ -203,35 +211,104 @@ def __init__(self, workflow, config, config_lookup_paths, config_paths, workdir)
+        sub_workflow = "somatic_variant_calling"
+        if config["is_filtered"]:
+            sub_workflow = "somatic_variant_filtration"
+        self.register_sub_workflow(sub_workflow, config["path_somatic_variant"], "somatic_variant")
         # Copy over "tools" setting from somatic_variant_calling/ngs_mapping if not set here
-        if not self.config.get("tools_ngs_mapping"):
-            self.config["tools_ngs_mapping"] = self.w_config["step_config"]["ngs_mapping"]["tools"][
+        if not config["tools_ngs_mapping"]:
+            config["tools_ngs_mapping"] = self.w_config["step_config"]["ngs_mapping"]["tools"][
                 "dna"
             ]
-        if not self.config.get("tools_somatic_variant_calling"):
-            self.config["tools_somatic_variant_calling"] = self.w_config["step_config"][
+        if not config["tools_somatic_variant_calling"]:
+            config["tools_somatic_variant_calling"] = self.w_config["step_config"][
                 "somatic_variant_calling"
             ]["tools"]
+        if not config["tools_somatic_variant_annotation"]:
+            config["tools_somatic_variant_annotation"] = self.w_config["step_config"][
+                "somatic_variant_annotation"
+            ]["tools"]
+        if config["is_filtered"]:
+            if len(self.w_config["step_config"]["somatic_variant_filtration"]["filter_list"]) > 0:
+                config["filters"] = []
+                config["filtered_regions"] = []
+            else:
+                if not config["filters"]:
+                    config["filters"] = list(
+                        self.w_config["step_config"]["somatic_variant_filtration"][
+                            "filter_sets"
+                        ].keys()
+                    )
+                    config["filters"].append("no_filter")
+                if not config["filtered_regions"]:
+                    config["filtered_regions"] = list(
+                        self.w_config["step_config"]["somatic_variant_filtration"][
+                            "exon_lists"
+                        ].keys()
+                    )
+                    config["filtered_regions"].append("genome_wide")
+        self.w_config["step_config"]["somatic_variant_signatures"] = config
+        # Register sub step classes so the sub steps are available
+        self.register_sub_step_classes(
+            (TabulateVariantsStepPart, DeconstructSigsStepPart, LinkOutStepPart)
+        )
 
     @listify
     def get_result_files(self):
         """Return list of result files for workflow"""
-        callers = set(self.config["tools_somatic_variant_calling"])
-        name_pattern = "{mapper}.{caller}.deconstruct_sigs.{tumor_library.name}"
+        config = self.w_config["step_config"]["somatic_variant_signatures"]
+        name_pattern = "{mapper}.{caller}"
+        if config["is_filtered"]:
+            if len(config["filters"]) > 0:
```

Comment:
tiny detail: in all cases beforehand the if-clause is the other way around starting with `if len(config["filters"]) == 0` . Meaning is always the same, so the phrase could also be the same.